src/olympia/devhub/tasks.py

```diff
@@ -414,16 +415,46 @@ def run_addons_linter(path, listed=True):
     if not listed:
         args.append('--self-hosted')
 
-    process = subprocess.Popen(
-        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if not os.path.exists(path):
+        raise ValueError(
+            'Path "{}" is not a file or directory or does not exist.'
+            .format(path))
+
+    file_size = os.path.getsize(path)
+
+    if file_size > settings.FILE_UPLOAD_MAX_MEMORY_SIZE:
```

Comment:
Sounds good, I'll change it
 
 
On Fri, May 20, 2016, at 11:44 AM, Mathieu Pillard wrote:
> In src/olympia/devhub/tasks.py[1]:


> > @@ -414,16 +415,46 @@ def run_addons_linter(path, listed=True): >
> > if not listed: >          args.append('--self-hosted') > > -
> > process = subprocess.Popen( > -        args, stdout=subprocess.PIPE,
> > stderr=subprocess.PIPE) > +    if not os.path.exists(path): > +
> > raise ValueError( > +            'Path "{}" is not a file or
> > directory or does not exist.' > +            .format(path)) > + > +
> > file_size = os.path.getsize(path) > + > +    if file_size >
> > settings.FILE_UPLOAD_MAX_MEMORY_SIZE:
>
> I think you are going to need your own setting with a more sensible
> value designed for the validator. FILE_UPLOAD_MAX_MEMORY_SIZE is
> probably too big.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub[2]
>
 

Links:

  1. https://github.com/mozilla/addons-server/pull/2715#discussion_r64015357
  2. https://github.com/mozilla/addons-server/pull/2715/files/e395f7d56b96c86735c023ff73a0517d957374f2#r64015357